RecoTauTag/RecoTau/src/PositionAtECalEntranceComputer.cc

```diff
@@ -2,37 +2,137 @@
 
 #include "MagneticField/Engine/interface/MagneticField.h"
 #include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
 #include "CommonTools/BaseParticlePropagator/interface/BaseParticlePropagator.h"
+#include "DataFormats/GeometrySurface/interface/Plane.h"
+#include "TrackPropagation/RungeKutta/interface/defaultRKPropagator.h"
 
 #include <cassert>
 
-PositionAtECalEntranceComputer::PositionAtECalEntranceComputer() : bField_z_(-1.) {}
+//HGCal helper classes
+//MB: looks be copy of HGCal utils: L1Trigger/L1THGCalUtilities/plugins/ntuples/HGCalTriggerNtupleGen.cc
+namespace hgcal_helpers {
+  class SimpleTrackPropagator {
+  public:
+    SimpleTrackPropagator(MagneticField const* f) : field_(f), prod_(field_, alongMomentum, 5.e-5), absz_target_(0) {
+      ROOT::Math::SMatrixIdentity id;
+      AlgebraicSymMatrix55 C(id);
+      C *= 0.001;
+      err_ = CurvilinearTrajectoryError(C);
+    }
+    void setPropagationTargetZ(const float& z);
+
+    bool propagate(const double& px,
+                   const double& py,
+                   const double& pz,
+                   const double& x,
+                   const double& y,
+                   const double& z,
+                   const float& charge,
+                   reco::Candidate::Point& output) const;
+
+  private:
+    SimpleTrackPropagator() : field_(nullptr), prod_(field_, alongMomentum, 5e-5), absz_target_(0) {}
```

Comment:
what is the meaning of 5e-5 here?

Reply:
As far as I understand this is tolerance with which point of an extrapolated trajectory lies on a surface to which it the trajectory is propagated. The 5e-5 is a default value (a small number corresponding with float precision?) and can be removed from this initialization list. Also `alongMomentum` direction is a default value and can be removed from the list.